gopherage/pkg/cov/junit/xmlwriter.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package takes coverage profile file as input and produces junit xml that is consumable by TestGrid
+
+package junit
+
+import (
+	"encoding/xml"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/tools/cover"
+
+	"k8s.io/test-infra/gopherage/pkg/cov/junit/calculation"
+)
+
+// Property defines the xml element that stores the value of code coverage
+type Property struct {
+	Name  string `xml:"name,attr"`
+	Value string `xml:"value,attr"`
+}
+
+// Properties defines the xml element that stores the a list of properties that associated with one testcase
+type Properties struct {
+	PropertyList []Property `xml:"property"`
+}
+
+// TestCase defines the xml element that stores all information associated with one test case
+type TestCase struct {
+	ClassName    string     `xml:"class_name,attr"`
+	Name         string     `xml:"name,attr"`
+	Time         string     `xml:"time,attr"`
+	Failure      bool       `xml:"failure,omitempty"`
+	PropertyList Properties `xml:"properties"`
+}
+
+// Testsuite defines the outer-most xml element that contains all test cases
+type Testsuite struct {
+	XMLName   string     `xml:"testsuite"`
+	Testcases []TestCase `xml:"testcase"`
+}
+
+// newTestCase constructs the TestCase struct
+func newTestCase(targetName, coverage string, failure bool) TestCase {
```

Comment:
nit: I think the `failure` parameter here is surprisingly inverted (passing `false` to mean success). I'd prefer to see a `success` parameter instead.